File: Algorithms/p005_Longest_Palindromic_Substring/p005_Longest_Palindromic_Substring.py
class Solution(object):
    def longestPalindrome(self, s):
        """
        :type s: str
        :rtype: str
        """
        n = len(s)
        start = 0
        length = 1
        isPalindrome = [[True]*n for i in range(n)]
        for i in range(1,n):
            for j in range(i+1):
                isPalindrome[j][i] = (s[i]==s[j]) and (j+1>i-1 or isPalindrome[j+1][i-1])
                if isPalindrome[j][i] and i-j+1>=length:
                    start = j
                    length = i-j+1
        return s[start:start+length]

File: Algorithms/p005_Longest_Palindromic_Substring/p005_Longest_Palindromic_Substring.cpp
class Solution {
public:
    string longestPalindrome(string s) {
        int n = s.size();
        int start = 0;
        int length = 1;
        vector<vector<int>> isPalindrome(n,vector<int>(n,1));
        for(int i=1;i<n;++i){
            for(int j=0;j<i;++j){
                isPalindrome[j][i] = (s[i]==s[j])&&(j+1>i-1||isPalindrome[j+1][i-1]);
                if(isPalindrome[j][i]&&i-j+1>length){
                    start = j;
                    length = i-j+1;
                }
            }
        }
        return s.substr(start,length);
    }
};
